### PR TITLE
BF: Fix range utility

### PR DESCRIPTION
**Pull request by LaurenSpiegel**
The setPartRanges function takes an outer range request and applies it to the parts in the data source.  We had some off by one errors which this PR fixes.

Formal review requested, @vrancurel

### CODE REVIEW

**Comment by rahulreddy** [comment]
👍

File: lib/routes/routesUtils.js [hunk 1]
```diff
@@ -230,15 +242,22 @@ const routesUtils = {
...
                     // and we're done
                 } else {
                     const partWithRange = dataLocations[i];
-                    const endPart = Math.min(partSize, max - total);
+                    // Need to subtranct one from endPart since range
+                    // includes endPart in byte count
+                    const endPart = Math.min(partSize - 1, max - total - 1);
                     partWithRange.range = [0, endPart];
+                    // modify size to be stored for object put part copy
+                    partWithRange.size =
```
Comment (alexandre-merle) [inline lib/routes/routesUtils.js]: can we use ```partWithRange.size = endPart + 1``` ? :)

File: lib/routes/routesUtils.js [hunk 2]
```diff
@@ -249,10 +268,17 @@ const routesUtils = {
                 // Use full remaining part if remaining partSize is less
                 // than byte range we need to satisfy.  Or use byte range
                 // we need to satisfy taking into account any startOffset
-                const endPart = Math.min(partSize, max - total + startOffset);
+                const endPart = Math.min(partSize - 1,
```
Comment (alexandre-merle) [inline lib/routes/routesUtils.js]: ```partSize - 1``` should not include startOffset too ?

File: lib/routes/routesUtils.js [hunk 3]
```diff
@@ -249,10 +268,17 @@ const routesUtils = {
...
+                    max - total + startOffset - 1);
                 partWithRange.range = [startOffset, endPart];
+                // modify size to be stored for object put part copy
+                partWithRange.size = Math.min(partSize,
```
Comment (alexandre-merle) [inline lib/routes/routesUtils.js]: ```endPart - startOffset + 1``` should be enought right ? :)

File: lib/routes/routesUtils.js [hunk 4]
```diff
@@ -230,15 +242,22 @@ const routesUtils = {
...
+                    const partWithoutRange = dataLocations[i];
+                    partWithoutRange.size = partSize.toString();
+                    parsedLocations.push(partWithoutRange);
                     total += partSize;
                     // Otherwise set a range limit on the part end
                     // and we're done
                 } else {
                     const partWithRange = dataLocations[i];
-                    const endPart = Math.min(partSize, max - total);
+                    // Need to subtranct one from endPart since range
```
Comment (alexandre-merle) [inline lib/routes/routesUtils.js]: ```substract``` :)

File: lib/routes/routesUtils.js (hunk 3 above)
Comment (LaurenSpiegel) [inline lib/routes/routesUtils.js]: yes, because I already did math.min in the endPart definition.  thanks

**Comment by LaurenSpiegel** [comment]
@ironman-machine range check

**Comment by LaurenSpiegel** [comment]
comments addressed

**Comment by alexandre-merle** [comment]
LGTM :+1:

**Pull request merged**